tensorflow/core/kernels/fused_batch_norm_op.h

```diff
@@ -43,6 +45,66 @@ struct InvVarianceToVariance {
   void operator()(const Eigen::GpuDevice& d, double epsilon, int sample_size,
                   int channels, T* variance);
 };
+
+
+// Functor used by FusedBatchNormGradOp to do the computations when is_training=False.
+// Both CPU and GPU will use this functor.
+template <typename Device, typename T>
+struct FusedBatchNormFreezeGrad {
```

Comment:
Could you move this function to fused_batch_norm_op.cc?

Reply:
If I understand the build process correctly, this functor needs to be instantiated in both fused_batch_norm_op.cc and fused_batch_norm_op.cu.cc, to be compiled to two kernels by nvcc and gcc respectively.  Therefore it has to be in a header file to be included by both .cc and .cu.cc. This seems like what's been done for other kernels as well.

Reply:
Sounds good.